### PR TITLE
Change event grid track1 to track2

### PR DESCRIPTION
**Pull request by Luyunmt**
Change event grid track1 to track2
@jongio for notification

### CODE REVIEW

**Comment by jongio** [comment]
@jsquire - Can you u please review?

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs [hunk 1]
```diff
@@ -29,19 +28,16 @@ public async Task<IActionResult> Post()
                 using (var reader = new StreamReader(Request.Body, Encoding.UTF8))
                 {
                     var jsonContent = await reader.ReadToEndAsync();
-                    var eventGridEvents = JsonConvert.DeserializeObject<List<EventGridEvent>>(jsonContent);
+                    var eventGridEvents = JsonSerializer.Deserialize<List<EventGridEvent>>(jsonContent);
```
Comment (jsquire) [inline src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: I believe that we want to prefer using `EventGridEvent.Parse` here.  _([ref](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/samples/Sample3_ParseAndDeserializeEvents.md#parse-events-from-json-payload))_

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs [hunk 2]
```diff
@@ -29,19 +28,16 @@ public async Task<IActionResult> Post()
...
 
                     foreach (var eventGridEvent in eventGridEvents)
                     {
                         // EventGrid validation message
-                        if (eventGridEvent.EventType == EventTypes.EventGridSubscriptionValidationEvent)
+                        if (eventGridEvent.EventType == "Microsoft.EventGrid.SubscriptionValidationEvent")
```
Comment (jsquire) [inline src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: It's a shame that [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventTypes.cs) was marked `internal`.    

@JoshLove-msft:  Any thoughts here?

**Review by jsquire — APPROVED** [review]

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs (hunk 2 above)
Comment (JoshLove-msft) [reply src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: we can do a type check as shown [here](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventgrid/Azure.Messaging.EventGrid#receiving-and-deserializing-events).
```cs
foreach (EventGridEvent egEvent in egEvents)
{
    // If the event is a system event, GetData() should return the correct system event type
   if (egEvent.GetData() is SubscriptionValidationEventData subscriptionData) 
  {
     return Ok(subscriptionData.ValidationCode);
  }
}
```

**Comment by JoshLove-msft** [comment]
Please keep in mind the EG library is still in beta, so the API may change which would mean we would need to update the samples again.

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs (hunk 2 above)
Comment (jsquire) [reply src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: Thanks!   I figured that I was overlooking something.

**Comment by Luyunmt** [comment]
Closed it due to commit (https://github.com/Azure-Samples/service-bus-java-manage-with-claims-based-authorization/commit/c21608ffc4993821f648986c1d0887f27f0f0c7c) contained this PR change.

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs [hunk 3]
```diff
@@ -29,22 +26,20 @@ public async Task<IActionResult> Post()
...
                         {
-                            var eventData = ((JObject)(eventGridEvent.Data)).ToObject<SubscriptionValidationEventData>();
-                            var responseData = new SubscriptionValidationResponse()
+                            // EventGrid validation message
+                            if (data is SubscriptionValidationEventData subscriptionData)
                             {
-                                ValidationResponse = eventData.ValidationCode
-                            };
-                            return Ok(responseData);
+                                return Ok(subscriptionData.ValidationCode);
```
Comment (JoshLove-msft) [inline src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: Does this work? I think we need to still create the SubscriptionValidationResponse.

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs [hunk 4]
```diff
@@ -58,7 +53,7 @@ public async Task<IActionResult> Post()
         }
         private async Task<IActionResult> HandleEvent(EventGridEvent eventGridEvent)
         {
-            if (eventGridEvent.EventType == EventTypes.StorageBlobCreatedEvent)
+            if (eventGridEvent.EventType == "Microsoft.Storage.BlobCreated")
```
Comment (JoshLove-msft) [inline src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: Can use SystemEventNames.StorageBlobCreated

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs (hunk 3 above)
Comment (Tianyun-G) [reply src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: @JoshLove-msft - I have updated, please review.

File: src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs (hunk 4 above)
Comment (Tianyun-G) [reply src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs]: @JoshLove-msft - I have updated, please review.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by v-xuto** [comment]
@jsquire  @JoshLove-msft  - Could you help us merge this PR?

**Comment by JoshLove-msft** [comment]
> @jsquire @JoshLove-msft - Could you help us merge this PR?

I don't think either one of us has permissions to do this. @jongio - can you help?

**Review by jongio — APPROVED** [review]

**Pull request merged**